src/BSON/Timestamp.c

```diff
@@ -214,6 +215,43 @@ PHP_METHOD(Timestamp, __toString)
 }
 /* }}} */
 
+/* {{{ proto array Timestamp::jsonSerialize()
+*/
+PHP_METHOD(Timestamp, jsonSerialize)
+{
+	php_phongo_timestamp_t *intern;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	intern = Z_TIMESTAMP_OBJ_P(getThis());
+
+	array_init_size(return_value, 1);
+
+#if PHP_VERSION_ID >= 70000
+	{
+		zval ts;
+
+		array_init_size(&ts, 2);
+		ADD_ASSOC_LONG_EX(&ts, "t", intern->timestamp);
+		ADD_ASSOC_LONG_EX(&ts, "i", intern->increment);
```

Comment:
After speaking with @llvtt, it's clear that the spec is not far enough along to make an assumption about the Timestamp format.

@derickr: Would you rather we throw an exception for out-of-range values on 32-bit systems or instead allow the unsigned values to be printed in their signed representation? I can re-purpose [PHPC-828](https://jira.mongodb.org/browse/PHPC-828) if necessary.